setup.sh

```diff
@@ -42,11 +42,21 @@ function _unset_vars
   unset SCRIPT USING_SELINUX
 }
 
-function _get_current_directory
+function _get_script_path
 {
-  if dirname "$(readlink -f "${0}")" &>/dev/null
+  if uname -a | grep -q Darwin
   then
-    CDIR="$(dirname "$(readlink -f "${0}")")"
+    SCRIPT_PATH="$(greadlink -f "${0}")"
```

Comment:
We should not use `greadlink`. We do not officially support macOS. The script _should_ work in a way which

1. Checks if `dirname "$(readlink -f "${0}")"` work, if not, go on
2. Checks if `dirname "$(readlink -f "${0}")"` works, if not, use the value provided on initialization, which is
3. The value of `$(pwd)`

Reply:
I agree. That was one of the reasons, I haven't approved this PR yet. I don't have a Mac to test with. That would imply, that neither `pwd` nor `realpath` is available in @mommel's environment, which sounds strange.

Reply:
`pwd` does exist on Big Sur, `realpath` does not.

Reply:
@fbartels Thanks for pointing this out! But as long as one of the three commands work, this should not be a problem in the end, i.e. I still have no clue as to why this should not work :)

Reply:
What about getting rid of the whole `_get_current_directory` function? 😄 

`CDIR=${0%/*}` should all we need. Pure Bash and no dependencies to other binarys.

Reply:
> What about getting rid of the whole `_get_current_directory` function? smile
> 
> `CDIR=${0%/*}` should all we need. Pure Bash and no dependencies to other binarys.

Quite nice, but only under the condition that working with relative paths is okay. I do not see an problems at the moment, but I'm not 100% confident either.

Reply:
Yes, I've considered this. In the current code, there is no need for an absolute path.

Reply:
> Yes, I've considered this. In the current code, there is no need for an absolute path.

Although we're "losing" the capability to resolve softlinks, this should not be an issue altogether. I will close this PR. Can you provide a new one which implements this behavior?

Reply:
Unfortunately, an absolute path is required for `docker -v` command :confused: